Benchmark/src/PWR032.cpp

```diff
@@ -0,0 +1,49 @@
+#include "Benchmark.h"
+
+// Forward-declare the functions to benchmark
+extern "C" {
+void example(const int N, const float *X1, const float *Y1, const float *X2,
+             const float *Y2, float *result);
+void solution(const int N, const float *X1, const float *Y1, const float *X2,
+              const float *Y2, float *result);
+}
+
+// Size adjusted to fit execution on micro-seconds
+constexpr int N = 128 * 128 * 128;
+
+#if OCB_ENABLE_C
+
+static void CExampleBench(benchmark::State &state) {
+  const auto &X1 = OpenCatalog::CreateRandomVector<float>(N);
+  const auto &Y1 = OpenCatalog::CreateRandomVector<float>(N);
+
+  const auto &X2 = OpenCatalog::CreateRandomVector<float>(N);
+  const auto &Y2 = OpenCatalog::CreateRandomVector<float>(N);
```

Comment:
nit:
```suggestion
  auto X1 = OpenCatalog::CreateRandomVector<float>(N);
  auto Y1 = OpenCatalog::CreateRandomVector<float>(N);
  auto X2 = OpenCatalog::CreateRandomVector<float>(N);
  auto Y2 = OpenCatalog::CreateRandomVector<float>(N);
```

Although it is allowed, I wouldn't take a `const` reference to a temporary object. For me, personally, it raises a lot of questions.

Reply:
Maybe I'm lacking some knowledge but as far as I know, your solution will apply `RVO` and mine will apply `NRVO`, I'll leave this comment open as maybe someone can provide more information about this topic.

Reply:
Both solutions are correct, indeed. However, and this is my opinion, taking a `const` reference to a temporary object (e.g., `const auto &foo = thisReturnsATemporary()`) raises a lot of questions. For instance, if you take a `const` reference in line 17, why wouldn't you take a reference to the temporary returned in lines 23 and 38? Again, both solutions are correct, but I found the `const auto &` thing hard to follow.